### ISSUE TITLE
What is the effect of mask_zero=True

### ISSUE THREAD

**Issue by lqj1990** [fchollet/keras#2897]
I want to use RNN to process a sequence padded with zeros, so I have to set mask_zero True in Embedding layer. But I don't know the expected outcome. I thought input 0 should get a zero vector after embedding if I set mask_zero True. But after I do some test, I don't see the difference between the True and False. Could anybody explain the effect of mask_zero=True?
I do some test using following code:
`X = [[0,1],[1,0],[0,0]];
model = Sequential();
model.add(Embedding(4,2,input_length=2,mask_zero=True));
model.compile(loss='mean_squared_error', optimizer=Adagrad(lr=0.01), metrics=['accuracy']);
out = model.predict(X);
print out;`
And the outcome is:
`[[[ 0.00132277 -0.04322269]
  [-0.00558603  0.04554391]]

 [[-0.00558603  0.04554391]
  [ 0.00132277 -0.04322269]]

 [[ 0.00132277 -0.04322269]
  [ 0.00132277 -0.04322269]]]`
